### PR TITLE
Set up SentryOptions to allow for skipping module registration

### PR DESCRIPTION
**Pull request by josh-degraw**
In reference to #152, this pull request adds ability to skip module registration on event processing.

### CODE REVIEW

File: src/Sentry/SentryOptions.cs [hunk 1]
```diff
@@ -320,6 +320,11 @@ public IDiagnosticLogger DiagnosticLogger
             }
         }
 
+        /// <summary>
+        /// Whether or not to include referenced assemblies in each event sent to sentry. Defaults to <see langword="true"/>.
+        /// </summary>
+        public virtual bool ReportAssemblies { get; set; } = true;
```
Comment (bruno-garcia) [inline src/Sentry/SentryOptions.cs]: Do you have a use-case to extend this property?
I'd rather leave it non virtual for the time being as Options passing was not designed to be extended.

**Comment by bruno-garcia** [comment]
Couldn't find you on Gitter so, unrelated to this PR but not sure you get the @ from the other repo: https://github.com/getsentry/sentry-docs/pull/993

File: src/Sentry/SentryOptions.cs (hunk 1 above)
Comment (josh-degraw) [reply src/Sentry/SentryOptions.cs]: Oh yeah, not sure why I set that like that. Updated.

**Comment by bruno-garcia** [comment]
@josh-degraw thanks again for helping!

**Pull request merged**